langchain/src/sql_database.ts

```diff
@@ -0,0 +1,130 @@
+import { DataSource, DataSourceOptions } from "typeorm";
+import {
+  generateTableInfoFromTables,
+  getTableAndColumnsName,
+  SerializedSqlDatabase,
+  SqlDatabaseDataSourceParams,
+  SqlDatabaseOptionsParams,
+  SqlTable,
+  verifyIgnoreTablesExistInDatabase,
+  verifyIncludeTablesExistInDatabase,
+  verifyListTablesExistInDatabase,
+} from "./util/sql_utils.js";
+
+export class SqlDatabase
+  implements SqlDatabaseOptionsParams, SqlDatabaseDataSourceParams
+{
+  appDataSourceOptions: DataSourceOptions;
+
+  appDataSource: DataSource;
+
+  allTables: Array<SqlTable> = [];
+
+  includesTables: Array<string> = [];
+
+  ignoreTables: Array<string> = [];
+
+  sampleRowsInTableInfo = 3;
+
+  protected constructor(fields: SqlDatabaseDataSourceParams) {
```

Comment:
made constructor protected bc we never want users instantiating this class with it (it wouldn't initialize the `DataSource` or `validateTables`). Those need to happen outside in a `from` static method bc they're async